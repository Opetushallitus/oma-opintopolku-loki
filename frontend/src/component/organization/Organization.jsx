import React from 'react'
import PropTypes from 'prop-types'
import { lensPath, view } from 'ramda'
import Expander from 'component/generic/widget/Expander'
import OrganizationDetails from 'component/organization/OrganizationDetails'
import LogEntries from 'component/log-entries/LogEntries'

/*
TODO: Currently we just take the first organization alternative (its name and oid). This must be changed.
 */
const nameLens = lensPath(['0', 'name'])

const title = organizations => view(nameLens, organizations)

const Organization = ({ organizationAlternatives, timestamps, serviceName, isMyDataUse, isJakolinkkiUse }) => (
  <Expander title={title(organizationAlternatives)} serviceName={serviceName}>
    <OrganizationDetails isMyDataUse={isMyDataUse} isJakolinkkiUse={isJakolinkkiUse}/>
    <LogEntries timestamps={timestamps}/>
  </Expander>
)

Organization.propTypes = {
  organizationAlternatives: PropTypes.array.isRequired,
  timestamps: PropTypes.array.isRequired,
  serviceName: PropTypes.string.isRequired,
  isMyDataUse: PropTypes.bool,
  isJakolinkkiUse: PropTypes.bool
}

export default Organization
