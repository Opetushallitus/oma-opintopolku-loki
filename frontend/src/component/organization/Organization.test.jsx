import React from 'react'
import { mount } from 'enzyme'
import Organization from './Organization'

describe('Organization', () => {
  const timestamps = ['2018-09-19T12:05:26.432+03', '2018-09-20T16:01:44.543+03']

  it('should render with organization permitted by default, and expand data when button is clicked', () => {
    const organizations = [{ oid: '1.1.111.111.11.11111111111', name: 'Organisaatio 1' }]
    const wrapper = mount(<Organization organizationAlternatives={organizations} timestamps={timestamps} />)
    expect(wrapper.render()).toMatchSnapshot()

    wrapper.find('button').at(0).simulate('click')
    expect(wrapper.render()).toMatchSnapshot()
  })

  it('should render with organization permitted by user, and expand data when button is clicked', () => {
    const organizations = [{ oid: '1.2.246.562.10.77876988401', name: 'Organisaatio 2' }]
    const wrapper = mount(<Organization organizationAlternatives={organizations} timestamps={timestamps} />)
    expect(wrapper.render()).toMatchSnapshot()

    wrapper.find('button').at(0).simulate('click')
    expect(wrapper.render()).toMatchSnapshot()
  })
})
