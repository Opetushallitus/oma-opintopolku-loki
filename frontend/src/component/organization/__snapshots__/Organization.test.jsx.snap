// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organization should render with organization permitted by default, and expand data when button is clicked 1`] = `
<div
  class="Expander__ExpanderContainer-sc-fukmjx-0 fExEJK"
>
  <button
    aria-pressed="false"
    class="Expander__ExpanderTitle-sc-fukmjx-2 ekUJgW"
  >
    <span
      aria-hidden="true"
      class="Expander__ExpanderPrefix-sc-fukmjx-1 cWphqs"
    >
      +
    </span>
    Organisaatio 1
  </button>
</div>
`;

exports[`Organization should render with organization permitted by default, and expand data when button is clicked 2`] = `
<div
  class="Expander__ExpanderContainer-sc-fukmjx-0 fExEJK"
>
  <button
    aria-pressed="true"
    class="Expander__ExpanderTitle-sc-fukmjx-2 gkGJNz"
  >
    <span
      aria-hidden="true"
      class="Expander__ExpanderPrefix-sc-fukmjx-1 cWphqs"
    >
      -
    </span>
    Organisaatio 1
  </button>
  <div
    class="Expander__ExpandedContents-sc-fukmjx-3 iNZtAp"
  >
    <div
      class="OrganizationDetails__Details-sc-4gjwy4-0 hGohEJ"
    >
      <div
        class="OrganizationDetails__Description-sc-4gjwy4-1 dxZCwD"
      >
        <b
          class="typography__Bold-sc-1yq3b4j-6 hlMPgm"
        >
          Tietojen käyttölupa:
        </b>
         Lakiin perustuva tiedonkäyttölupa.
      </div>
      <a
        class="typography__Link-sc-1yq3b4j-5 cJIYQp"
        href="https://opintopolku.fi/wp/fi/tietosuojaseloste/"
        target="_blank"
      >
        <test-file-stub
          classname="ExternalLink__PositionedLinkIcon-sc-1in9m5g-0 dRZGgw"
        />
        <b
          class="typography__Bold-sc-1yq3b4j-6 hlMPgm"
        >
          Tarkempi kuvaus lähetetyistä tiedoista
        </b>
      </a>
    </div>
    <div
      class="LogEntries__Container-sc-ufvf4x-0 bWYhUd"
    >
      <div
        class="LogEntries__TopRow-sc-ufvf4x-1 cVnINA"
      >
        <button
          aria-pressed="false"
          class="ButtonSmall__TextButton-sc-r8zxw8-0 jPkZlO"
        >
          Näytä tietojen käyttökerrat
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Organization should render with organization permitted by user, and expand data when button is clicked 1`] = `
<div
  class="Expander__ExpanderContainer-sc-fukmjx-0 fExEJK"
>
  <button
    aria-pressed="false"
    class="Expander__ExpanderTitle-sc-fukmjx-2 ekUJgW"
  >
    <span
      aria-hidden="true"
      class="Expander__ExpanderPrefix-sc-fukmjx-1 cWphqs"
    >
      +
    </span>
    Organisaatio 2
  </button>
</div>
`;

exports[`Organization should render with organization permitted by user, and expand data when button is clicked 2`] = `
<div
  class="Expander__ExpanderContainer-sc-fukmjx-0 fExEJK"
>
  <button
    aria-pressed="true"
    class="Expander__ExpanderTitle-sc-fukmjx-2 gkGJNz"
  >
    <span
      aria-hidden="true"
      class="Expander__ExpanderPrefix-sc-fukmjx-1 cWphqs"
    >
      -
    </span>
    Organisaatio 2
  </button>
  <div
    class="Expander__ExpandedContents-sc-fukmjx-3 iNZtAp"
  >
    <div
      class="OrganizationDetails__Details-sc-4gjwy4-0 hGohEJ"
    >
      <div
        class="OrganizationDetails__Description-sc-4gjwy4-1 dxZCwD"
      >
        <b
          class="typography__Bold-sc-1yq3b4j-6 hlMPgm"
        >
          Tietojen käyttölupa:
        </b>
         Olet antanut tälle palvelutarjoajalle luvan käyttää tietojasi.
      </div>
      <a
        class="typography__Link-sc-1yq3b4j-5 cJIYQp"
        href="https://confluence.csc.fi/pages/viewpage.action?pageId=76536741"
        target="_blank"
      >
        <test-file-stub
          classname="ExternalLink__PositionedLinkIcon-sc-1in9m5g-0 dRZGgw"
        />
        <b
          class="typography__Bold-sc-1yq3b4j-6 hlMPgm"
        >
          Tarkempi kuvaus lähetetyistä tiedoista
        </b>
      </a>
    </div>
    <div
      class="LogEntries__Container-sc-ufvf4x-0 bWYhUd"
    >
      <div
        class="LogEntries__TopRow-sc-ufvf4x-1 cVnINA"
      >
        <button
          aria-pressed="false"
          class="ButtonSmall__TextButton-sc-r8zxw8-0 jPkZlO"
        >
          Näytä tietojen käyttökerrat
        </button>
      </div>
    </div>
  </div>
</div>
`;
